cmd/data-usage-cache.go

```diff
@@ -1080,6 +1189,33 @@ func (d *dataUsageCache) deserialize(r io.Reader) error {
 			d.Cache[k] = e
 		}
 		return nil
+	case dataUsageCacheVerV6:
+		// Zstd compressed.
+		dec, err := zstd.NewReader(r, zstd.WithDecoderConcurrency(2))
+		if err != nil {
+			return err
+		}
+		defer dec.Close()
+		dold := &dataUsageCacheV6{}
+		if err = dold.DecodeMsg(msgp.NewReader(dec)); err != nil {
+			return err
+		}
+		d.Info = dold.Info
+		d.Disks = dold.Disks
+		d.Cache = make(map[string]dataUsageEntry, len(dold.Cache))
+		for k, v := range dold.Cache {
+			due := dataUsageEntry{
+				Children:         v.Children,
+				Size:             v.Size,
+				Objects:          v.Objects,
+				Versions:         v.Versions,
+				ObjSizes:         v.ObjSizes,
+				ReplicationStats: v.ReplicationStats,
+			}
+			due.Compacted = len(due.Children) == 0 && k != d.Info.Name
```

Comment:
We shouldn't need to infer this (and will be wrong).
```suggestion
			due.Compacted = v.Compacted
```
(maybe just move it up, no biggie)